### PR TITLE
ReaderHighlight: set default long-press action with a gesture

### PR DESCRIPTION
**Pull request by hius07**
(1) Add "Add note" default 'long-press on text' action
(2) Allow setting the default action with a gesture (will help a little bit to https://github.com/koreader/koreader/issues/9695).

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9723)

### CODE REVIEW

File: frontend/apps/reader/modules/readerhighlight.lua [hunk 1]
```diff
@@ -20,7 +20,30 @@ local C_ = _.pgettext
...
+        "select",
+        "note",
+        "translate",
+        "wikipedia",
+        "dictionary",
+        "search",
+    },
+    long_press_action_text = {
+        _("Ask with popup dialog"),
+        _("Do nothing"),
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerhighlight.lua:37]: Any real reason to separate these value/text mapping into 2 tables that we'll have to not mess their sync ?

File: frontend/apps/reader/modules/readerhighlight.lua [hunk 2]
```diff
@@ -313,17 +336,6 @@ local note_mark = {
     {_("Side mark"), "sidemark"},
 }
 
-local long_press_action = {
-    {_("Ask with popup dialog"), "ask"},
-    {_("Do nothing"), "nothing"},
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerhighlight.lua:318]: Why not just move this one where you put the 2 new ones (and invert the values if it makes more sense, `{"ask", _("Ask with popup dialog") }` ?

File: frontend/apps/reader/modules/readerhighlight.lua [hunk 3]
```diff
@@ -490,16 +502,15 @@ The interval value is in seconds and can range from 3 to 20 seconds.]]),
...
         },
     }
-    for _, v in ipairs(long_press_action) do
+    for i, v in ipairs(self.long_press_action) do
         table.insert(menu_items.long_press.sub_item_table, {
-            text = v[1],
+            text = self.long_press_action_text[i],
             checked_func = function()
-                return G_reader_settings:readSetting("default_highlight_action", "ask") == v[2]
+                return v == G_reader_settings:readSetting("default_highlight_action", "ask")
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerhighlight.lua:509]: It indeed might feel a bit convoluted these `v[1]` and `v[2]`, but that's how we do mostly everywhere else.
You could just use `local name, text = v[1], v[2]` if you want it more readable.

File: frontend/apps/reader/modules/readerhighlight.lua (hunk 1 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerhighlight.lua:37]: For the purpose of Dispatcher.

Comment (poire-z) [reply frontend/apps/reader/modules/readerhighlight.lua:37]: OK, I see.
I would have probably made it be `toggle=ReaderHighlight.get_long_press_action_text()`, a function returning the list of text from the old 2-item mappings.

I remember there's been a few discussions about sometime adding functionality for plugins to be able to register some new highlight action to this list (mostly to the button table), although none do that currently. I guess we may have to revisit all that - and it's Dispatcher integration - when the times comes we have to add that :)

Comment (hius07) [reply frontend/apps/reader/modules/readerhighlight.lua]: Done

**Comment by hius07** [comment]
Unfortunately we now have 11 items in the "Long-press on text" menu.

**Review by poire-z — APPROVED** [review]

**Comment by poire-z** [comment]
> Unfortunately we now have 11 items in the "Long-press on text" menu.

I guess that's ok. It's not a top level menu, and not something we (at least I) visit often.
Also, I think the last one (Fulltext search) might be the least candidate for a default action :)

**Comment by hius07** [comment]
Agree.

Further to https://github.com/koreader/koreader/issues/9695, I think of a profile called "Disable touch".
Not so much, actually "Disable tap and long-press, but keep swipe". It would include:
- "Do nothing" for long-press on text
- "With swipes" only for page turns
- "With a swipe" only for activate menu
- zero size corner tap zones
- "Hold footer to skim" disabled

**Comment by NiLuJe** [comment]
I... hadn't really thought of that. That's the long way 'round, but it'll probably do the job, yeah ;).

**Pull request merged**